src/main/java/com/cleanroommc/groovyscript/compat/mods/lightningcraft/Infusion.java

```diff
@@ -0,0 +1,133 @@
+package com.cleanroommc.groovyscript.compat.mods.lightningcraft;
+
+import com.cleanroommc.groovyscript.GroovyScriptConfig;
+import com.cleanroommc.groovyscript.api.GroovyLog;
+import com.cleanroommc.groovyscript.api.IIngredient;
+import com.cleanroommc.groovyscript.api.documentation.annotations.*;
+import com.cleanroommc.groovyscript.compat.mods.ModSupport;
+import com.cleanroommc.groovyscript.helper.recipe.AbstractRecipeBuilder;
+import com.cleanroommc.groovyscript.registry.StandardListRegistry;
+import net.minecraft.item.ItemStack;
+import org.jetbrains.annotations.Nullable;
+import sblectric.lightningcraft.api.recipes.LightningInfusionRecipe;
+import sblectric.lightningcraft.recipes.LightningInfusionRecipes;
+
+import java.util.Collection;
+
+@RegistryDescription
+public class Infusion extends StandardListRegistry<LightningInfusionRecipe> {
+
+    @Override
+    public Collection<LightningInfusionRecipe> getRecipes() {
+        return LightningInfusionRecipes.instance().getRecipeList();
+    }
+
+    @MethodDescription(example = @Example("item('minecraft:diamond')"))
+    public boolean removeByOutput(IIngredient output) {
+        return getRecipes().removeIf(r -> output.test(r.getOutput()) && doAddBackup(r));
+    }
+
+    @RecipeBuilderDescription(example = {
+        @Example(".centerItem(item('minecraft:clay')).input(item('minecraft:gold_ingot'), item('minecraft:gold_ingot'), item('minecraft:iron_ingot'), item('minecraft:iron_ingot')).output(item('minecraft:nether_star')).le(500)"),
+        @Example(".centerItem(item('minecraft:clay')).input(item('minecraft:gold_ingot'), item('minecraft:potion').withNbt(['Potion': 'minecraft:leaping'])).output(item('minecraft:diamond_block')).le(200)"),
+    })
+    public RecipeBuilder recipeBuilder() {
+        return new RecipeBuilder();
+    }
+
+    @Property(property = "input", comp = @Comp(gte = 0, lte = 4))
+    @Property(property = "output", comp = @Comp(eq = 1))
+    public static class RecipeBuilder extends AbstractRecipeBuilder<LightningInfusionRecipe> {
+
+        @Property()
+        private IIngredient centerItem = null;
+
+        @Property(comp = @Comp(gte = 0), defaultValue = "-1")
+        private int le = -1;
+
+        @Property(defaultValue = "determined automatically based on the input items")
+        private boolean nbtSensitive = false;

```

Comment:
is there a reason why having this be directly configurable by the player is required? if it can be automatically determined, why not simply not expose it.

Reply:
Because there are cases where it's automatically configured value is wrong. For example, if you want to allow only input item that has no NBT in it (say, it's some tool that stores durability in NBT). You need to enable nbtSensitive = true, and there's not really another way to do it